### ISSUE TITLE
letsencrypt certificate recognized als self-signed

### ISSUE THREAD

**Issue by streaps** [cmeng-git/atalk-android#8]
I'm using a letsencrypt cetificate with my private prosody server. atalk issues a warning that it is a self-signed certificate. With other clients this is not happening, e.g. Conversations and AstraChat accept it as a regular cert.

**Comment by streaps** [comment]
It's fine to have a warning for self-signed certificates. But my server does not provide a self-signed certificate, it's signed by letsencrypt. There should be no warning. 

Three additional things:
- if I wait to long to to press proceed there is an error message (Timeout … 10s, while waitingfor establishing TLS). It took me a while to understand why it happens. 

- the permissions dialogue often pops up over "Certicate Verification" and when I press "done" the "Login User Credentials"  is shown with the timeout error.

- "Always trust this certificate" is somehow hidden under "Show Certificate" and then I have to scroll all the way down. I discovered it by chance. I think it would be better to put "Always trust this certificate" in the first popup.

There are lots of things happening in parallel and it could be very confusing for the user. My first impression was: everything is broken and nothing works.

**Comment by cmeng-git** [comment]
Sorry for the confusion. Currently aTalk uses multi-thread implementation, as there are so many tasks need to be executed on first launch. I will takes a look if I can do some things to make it less confusing.
aTalk version 1.3.0 implementation requires a server cert for security verification. Next aTalk version will have DNSSEC security implemented, I am not sure if this can help to solve your problem.

I feel strange to ask user to always trust a certificate even before reading it.

You want to visit the aTalk online help. I continuous update it with feedback from users. 
http://atalk.sytes.net/index.html

**Comment by streaps** [comment]
I've added a user account for testing

cmeng@opendialogue.eu
155d7d80f0a72595

my jid is
oku@opendialogue.eu

**Comment by cmeng-git** [comment]
Thanks, I have performed tests on aTalk using your account provided on both my Note-3 and Note-8. On Note-3, the account is able to log in, and there is no SSL certification verification prompt being issued from aTalk.

However on Note-8, user is prompted to confirm the SSL certificate signed by letsencrypt. Look like android-8 has excluded letsencrypt from their trusted CA list. You may refer to the following site for more information.
https://developer.android.com/training/articles/security-ssl.

Other than this, I do not detect any login problem based on the debug log on both Note-3 and Note-8.

**Comment by streaps** [comment]
I'm using LineageOS 14., which is Android 7.1.

Do you have any idea, why this doesn't happen with other xmpp clients? I tested the following clients and none of them issued a warning.

Android: Conversation, AstraChat, Tigase, Zom, Xabber
iOS: Monal IM, AstraChat
Linux: Psi, Pidgin, Gajim

**Comment by cmeng-git** [comment]
There are various approaches in implementing Connection Security Establishment. Please refer to the following site and the proposed links for better understanding on DNS security implementation. I am unable to advice you why other android clients does not reject your CA certificate without knowing their individual client actual implementation. May be you can revert to your CA for an explanation why android-8 excludes them in the android-8 OS trusted list.

[20180526] TLS Certificate, DNSSEC and DANE Security Implementation for Connection Establishment
http://atalk.sytes.net/faq.html#account_04

**Comment by streaps** [comment]
Seriously?

[Let's Encrypt](https://letsencrypt.org) is used everywhere, even by Google Firebase for custom domains.

The xmpp server jabber.org and conversations.im use Let's Encrypt certs, too and atalk also shows a warning. I'm pretty sure it is an atalk bug.

**Comment by cmeng-git** [comment]
I am reviewing the source, trying to understand why aTalk works on Note3 but reject the cert on Note8. Will come back to you once I have the answer.

**Comment by cmeng-git** [comment]
I believe I have solved the problem:
https://www.javatips.net/api/org.skife.config.configsource

With reference to the above link. Although the Network Security Configuration allows app to specify a trusted custom service in android#networkSecurityConfig, RootTrustManager#checkServerTrusted() is not happy when it finds that there are pre-configured trusted domains being specified; hence it throws the exception and returns CA certificate as not trusted. I have removed the custom trusted service, and seems working now. However I was unable to test on your server, the test account seems to have been disabled.

Also when I issue the following checking for ssl certficate details, the server returns error. Is your server support SSL via port 443?

```
openssl s_clientopenssl s_client -connect opendialogue.eu:443
CONNECTED(00000003)
140716427878848:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:../ssl/record/rec_layer_s3.c:1399:SSL alert number 40
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 176 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1527554516
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
---
```

**Comment by streaps** [comment]
You can check the TLS handshake with some non-registered username. If you didn't get any certificate warning and only an authentication error, it should work.

**Comment by cmeng-git** [comment]
Problem resolved in aTalk release version 1.3.1
